Source/FieldSolver/ImplicitSolvers/ImplicitSolverLibrary.H

```diff
@@ -0,0 +1,13 @@
+/* Copyright 2024 Justin Angus
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPLICIT_SOLVER_LIBRARY_H_
+#define IMPLICIT_SOLVER_LIBRARY_H_
+
+#include "SemiImplicitEM.H"
+#include "ThetaImplicitEM.H"
```

Comment:
We did not create facade headers extensively in WarpX yet, but it makes total sense here.

What you can do to make code maintenance tools recognize it better is [add these comments](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUPragmas.md#iwyu-pragma-export):
```suggestion
#include "SemiImplicitEM.H"  // IWYU pragma: export
#include "ThetaImplicitEM.H"  // IWYU pragma: export
```

Reply:
The `..Solver/...SolverLibrary.H` name appears a bit verbose.
Maybe we can name them something like `ImplicitSolvers/Library.H`?

Reply:
Related to the other comment in `WarpX.H`:
We could also add a file for [forward declarations](https://warpx.readthedocs.io/en/latest/developers/repo_organization.html#c-includes) `_fwd.H` to speed up the common use of `WarpX.H`.

Reply:
Any concern with having two different header files with the exact same name?

What does // IWYU pragma: export do?

Reply:
I read the links about forward declarations headers. Makes sense. Will do it here.